hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HoodieHiveClient.java

```diff
@@ -134,6 +136,41 @@ public void updateTableProperties(String tableName, Map<String, String> tablePro
     }
   }
 
+  /**
+   * Update the table serde properties to the table.
+   */
+  @Override
+  public void updateTableSerDeInfo(String tableName, String serdeClass, Map<String, String> serdeProperties) {
+    if (serdeProperties == null || serdeProperties.isEmpty()) {
+      return;
+    }
+    try {
+      Table table = client.getTable(syncConfig.databaseName, tableName);
+      serdeProperties.put("serialization.format", "1");
+      StorageDescriptor storageDescriptor = table.getSd();
+      SerDeInfo serdeInfo = storageDescriptor.getSerdeInfo();
+      if (serdeInfo != null && serdeInfo.getParametersSize() == serdeProperties.size()) {
+        Map<String, String> parameters = serdeInfo.getParameters();
+        boolean same = true;
+        for (String key : serdeProperties.keySet()) {
+          if (!parameters.containsKey(key) | !parameters.get(key).equals(serdeProperties.get(key))) {
+            same = false;
+            break;
+          }
+        }
+        if (same) {
+          LOG.debug("Table " + tableName + " serdeProperties already up to date, skip update");
+          return;
+        }
+      }
+      storageDescriptor.setSerdeInfo(new SerDeInfo(null, serdeClass, serdeProperties));
+      client.alter_table(syncConfig.databaseName, tableName, table);
```

Comment:
should we be calling       table.setSd(storageDescriptor); before calling client.alterTable ?